app/controllers/application_controller.rb

```diff
@@ -1,8 +1,25 @@
 class ApplicationController < ActionController::Base
   include Authenticate
 
+  def default_render
```

Comment:
This won't be called if we redirect within the action itself, so only _some_ actions will get JSON payloads in their response, namely those that would otherwise render from an HTML template.

Reply:
I figured out a way to ONLY include user created instance variables in the response. A simple before_action to remove anything that existed before the action ran.